lib/log/json_formatter.rb

```diff
@@ -0,0 +1,26 @@
+require 'json'
+
+module Log
+  class JSONFormatter
+
+    def call(severity, time, progname, msg)
+      log_item = {
+        app: progname,
+        host: host,
+        level: severity,
+      }
+      log_item[:message] = msg[:message] if msg[:message]
```

Comment:
You could simply merge the content of the hash into log_item. Let's discuss this tomorrow.

-----Original Message-----
From: "Peter Marton" <notifications@github.com>
Sent: ‎2015.‎11.‎11. 18:51
To: "sspinc/csv2avro" <csv2avro@noreply.github.com>
Cc: "Tamás Michelberger" <tomi@secretsaucepartners.com>
Subject: Re: [csv2avro] [WIP] Metrics and structured logging (#30)

In lib/log/json_formatter.rb:
> @@ -0,0 +1,26 @@
> +require 'json'
> +
> +module Log
> +  class JSONFormatter
> +
> +    def call(severity, time, progname, msg)
> +      log_item = {
> +        app: progname,
> +        host: host,
> +        level: severity,
> +      }
> +      log_item[:message] = msg[:message] if msg[:message]
I like the idea, but msg is a hash that can have multiple things at once. It's completely valid to have an event and corresponding metrics. So I don't see how we could generalize this more.
—
Reply to this email directly or view it on GitHub.